src/requirements/Github/GithubRequirement.tsx

```diff
@@ -1,32 +1,146 @@
 import { Icon } from "@chakra-ui/react"
-import Link from "components/common/Link"
 import ConnectRequirementPlatformButton from "components/[guild]/Requirements/components/ConnectRequirementPlatformButton"
+import DataBlockWithDate from "components/[guild]/Requirements/components/DataBlockWithDate"
+import RelativeDataBlockDate from "components/[guild]/Requirements/components/RelativeDataBlockDate"
 import Requirement, {
   RequirementProps,
 } from "components/[guild]/Requirements/components/Requirement"
 import { useRequirementContext } from "components/[guild]/Requirements/components/RequirementContext"
+import Link from "components/common/Link"
 import { GithubLogo } from "phosphor-react"
 
 const GithubRequirement = (props: RequirementProps) => {
   const requirement = useRequirementContext()
 
   return (
-    <Requirement
-      image={<Icon as={GithubLogo} boxSize={6} />}
-      footer={<ConnectRequirementPlatformButton />}
-      {...props}
-    >
-      Give a star to the{" "}
-      <Link
-        href={requirement.data.id ?? ""}
-        isExternal
-        colorScheme="blue"
-        fontWeight="medium"
-      >
-        {requirement.data.id.match(/https:\/\/github\.com\/(.+)$/i)[1]}
-      </Link>{" "}
-      repository
-    </Requirement>
+    <>
+      {requirement.type === "GITHUB_STARRING" ? (
+        <Requirement
```

Comment:
These lines
```tsx
 <Requirement
      image={<Icon as={GithubLogo} boxSize={6} />}
      footer={<ConnectRequirementPlatformButton />}
      {...props}
>
```
are repeated, you could just leave them at top-level, and only put the variable parts inside the if-else. Also, maybe a switch-case would be easier to read here? You can check our other requirement components, we've used switch-case inside those ;)